src/LanguageServer.php

```diff
@@ -215,10 +215,18 @@ public function initialize(ClientCapabilities $capabilities, string $rootPath =
                     sortUrisLevelOrder($composerLockFiles);
 
                     if (!empty($composerLockFiles)) {
-                        $this->composerLock = json_decode(yield $this->contentRetriever->retrieve($composerLockFiles[0]));
+                        $packages = [];
+                        $packagesDev = [];
+                        foreach ($composerLockFiles as $composerLockFile) {
+                            $composerLock = json_decode(yield $this->contentRetriever->retrieve($composerLockFile));
+                            $packages = array_merge($packages, $composerLock->packages);
+                            $packagesDev = array_merge($packagesDev, $composerLock->{'packages-dev'});
+                        }
+                        $this->composerLock = $composerLock;
+                        $this->composerLock->packages = $packages;
+                        $this->composerLock->{'packages-dev'} = $packagesDev;
```

Comment:
Hmm, are you sure aggregating all packages work? What if the same package appears with different versions, and then the cache key that is used for the index is not actually the package version that was indexed?